### PR TITLE
feat(command): Implement PUT /device/name/{deviceName}/{command} V2 API

### LINKED ISSUE #3176: [V2 Command]  Implement PUT /device/name/{deviceName}/{command} V2 API
# 🚀 Feature Request

### Relevant Package

 This feature request is for Core command

### Description

Per https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-command/2.x#/default/put_device_name__name___command_ implement PUT /device/name/{deviceName}/{command} V2 API

### PR DESCRIPTION
**Pull request by judehung**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/master/.github/Contributing.md.

## Issue Number:
fix #3176 

## What is the new behavior?
Per
https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-command/2.x#/default/put_device_name__name___command_
implement PUT /device/name/{deviceName}/{command} V2 API.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: internal/core/command/v2/controller/http/command.go [hunk 1]
```diff
@@ -127,3 +127,39 @@ func (cc *CommandController) IssueGetCommandByName(w http.ResponseWriter, r *htt
...
+	commandName := vars[v2.Command]
+
+	// Query params
+	queryParams := r.URL.RawQuery
+
+	var response commonDTO.BaseResponse
+	var statusCode int
+
+	// Request body
+	settings, err := utils.ParseBodyToMap(r)
```
[the remark this answers is not in the document]
Comment (judehung) [reply internal/core/command/v2/controller/http/command.go:148]: @lenny-intel the inter-service communication relies on client library, and the V2 client library expects to receive DTO structs and then do json encoding when issuing requests to target services: https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/v2/clients/http/utils/common.go#L84
If we pass through Body as []byte, client library will encode the Body []byte as a base64-encoded string per [json.Marshal](https://golang.org/pkg/encoding/json/#Marshal), thus failing the Body parsing logic later at device service sdk.

Per https://github.com/edgexfoundry/edgex-docs/blob/master/docs_src/design/adr/device-service/0011-DeviceService-Rest-API.md#device ADR, the body is expected to be a set of key/value pairs, thus a map[string]string should be sufficient to represent the Body, as we don't define specific DTO to represent the Body.

One advantage of this implementation is that Command Service could directly return error to invokers if Body is not in valid key/value pairs format, as opposed to passing invalid format of Body to client library and then waiting for error response.

**Comment by judehung** [comment]
rebased

File: internal/core/command/v2/controller/http/command.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (lenny-intel) [reply internal/core/command/v2/controller/http/command.go:148]: Got it. Thanks for the detailed explanation! :-)

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**